iceoryx_utils/source/posix_wrapper/file_lock.cpp

```diff
@@ -0,0 +1,267 @@
+            return cxx::error<FileLockError>(convertErrnoToFileLockError(openCall.getErrNum()));
+        }
+        else if (lockCall.getErrNum() == EWOULDBLOCK)
+        {
+            closeFileDescriptor();
+            // possible errors in closeFileDescriptor() are masked and we inform the user about the actual error
+            return cxx::error<FileLockError>(FileLockError::LOCKED_BY_OTHER_PROCESS);
+        }
+    }
+    return cxx::success<>();
+}
+
+FileLock::FileLock(FileLock&& rhs) noexcept
+{
+    *this = std::move(rhs);
+}
+
+FileLock& FileLock::operator=(FileLock&& rhs) noexcept
+{
+    if (this != &rhs)
+    {
+        if (closeFileDescriptor().has_error())
+        {
+            std::cerr << "Unable to cleanup file lock \"" << m_name
+                      << "\" in the move constructor/move assingment operator" << std::endl;
+        }
+
+        CreationPattern_t::operator=(std::move(rhs));
+
+        m_name = std::move(rhs.m_name);
+        m_fd = std::move(rhs.m_fd);
+
+        rhs.m_name.assign("");
+        rhs.m_fd = INVALID_FD;
+    }
+
+    return *this;
+}
+
+FileLock::~FileLock() noexcept
+{
+    if (closeFileDescriptor().has_error())
+    {
+        std::cerr << "unable to cleanup file lock \"" << m_name << "\" in the destructor" << std::endl;
+    }
+}
+
+cxx::expected<FileLockError> FileLock::closeFileDescriptor() noexcept
+{
+    if (isInitialized() && (m_fd != INVALID_FD))
+    {
+        auto closeCall = cxx::makeSmartC(iox_close, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {ERROR_CODE}, {}, m_fd);
+
+        if (!closeCall.hasErrors())
+        {
+            m_fd = INVALID_FD;
+            m_isInitialized = false;
+
+            return cxx::success<void>();
+        }
+        else
+        {
+            return cxx::error<FileLockError>(convertErrnoToFileLockError(closeCall.getErrNum()));
+        }
+    }
+    return cxx::success<>();
+}
+
+FileLockError FileLock::convertErrnoToFileLockError(const int32_t errnum) const noexcept
+{
+    switch (errnum)
+    {
+    case EACCES:
+    {
+        std::cerr << "permission to access file denied \"" << m_name << "\"" << std::endl;
+        return FileLockError::ACCESS_DENIED;
+    }
+    case EDQUOT:
+    {
+        std::cerr << "user disk quota exhausted for file \"" << m_name << "\"" << std::endl;
+        return FileLockError::QUOTA_EXHAUSTED;
+    }
+    case EEXIST:
+    {
+        std::cerr << "file \"" << m_name << "\""
+                  << " already exists" << std::endl;
+        return FileLockError::FILE_EXISTS;
+    }
+    case EFAULT:
+    {
+        std::cerr << "outside address space error for file \"" << m_name << "\"" << std::endl;
+        return FileLockError::ACCESS_DENIED;
+    }
+    case EFBIG:
+    case EOVERFLOW:
+    {
+        std::cerr << "file \"" << m_name << "\""
+                  << " is too large to be openend" << std::endl;
+        return FileLockError::FILE_TOO_LARGE;
+    }
+    case EINVAL:
```

Comment:
This errno should never occur. Invalid arguments can be avoided by the abstraction!
If it is occurring then we have an error in our abstraction and therefore we should always print the internal logic error.

Could you please remove this as well as the `FileLockError::INVALID_ARGUMENTS`.

Reply:
Please try to resist from doing another review before reading the comments to the previous review, thanks!

[Here's](https://github.com/eclipse-iceoryx/iceoryx/pull/606#discussion_r594479769) the answer. Furthermore, we can't know every possible mounting layout beforehand. To avoid confusion I renamed the error to `INVALID_PARAMETERS`.